File: CarbonaraWebAPI/Model/DAO/Car.cs
using System.ComponentModel.DataAnnotations;

namespace CarbonaraWebAPI.Model.DAO
{
    public class Car
    {

        public enum CarStatus { onStation, moving, parked , unavailable };
        public enum CarLockStatus { unlocked, locked };

        public Car( int kilometersDriven, float tankLevel, CarStatus status, CarLockStatus lockStatus, Cartype type , string licensePlateNumber)
        {
            LicensePlateNumber = licensePlateNumber;
            KilometersDriven = kilometersDriven;
            TankLevel = tankLevel;
            Status = status;
            LockStatus = lockStatus;
            Type = type;
        }

        public Car()
        {
        }

        [Key]
        public int Id { get; set; }
        [Required]
        public int KilometersDriven { get; set; } = 0;
        [Required]
        public float TankLevel { get; set; } = 0;
        [Required]
        public string LicensePlateNumber { get; set; }
        [Required]
        public CarStatus Status { get; set; }
        [Required]
        public CarLockStatus LockStatus { get; set; }
        [Required]
        public Cartype Type { get; set; }
    }
}

File: CarbonaraAngular/src/app/model/car.ts
import { Cartype } from "./cartype";

export class Car {
    constructor(

        public id: number,
        public kilometersDriven: number,
        public tankLevel: number,
        public licensePlateNumber: string,
        public status: CarStatus,
        public lockStatus: LockStatus,
        public type: Cartype,
    ) {

    }
   
}

export enum CarStatus { 
    onStation, 
    moving, 
    parked , 
    unavailable
}

export enum LockStatus { 
    unlocked, 
    locked
}